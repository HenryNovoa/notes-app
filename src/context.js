import React from 'react'

export const NotesContext = React.createContext({
    currentNote: null,
    notes: [
        {id: 1, text: 'Do small assestment'},
        {id: 2, text: 'Go grocery shopping'}
    ]
})

