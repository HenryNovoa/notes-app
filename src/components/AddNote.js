import { useState, useContext, useRef, useEffect } from 'react'
import { NotesContext } from '../context'

export function AddNote() {
    const { dispatch } = useContext(NotesContext)
    const [ value, setValue ] = useState('')

    let ref = useRef()

    useEffect(()=> {
        ref.current.focus();
    }, [])

    const handleChange = (event) => {
        setValue(event.target.value)
    }

    const handleSubmit = (event) => {
        event.preventDefault()
        if(value.trim === '') return alert('Cannot add a blank note')

        dispatch({type: 'ADD_NOTE', payload: value})
        setValue('')
    }
    return (
        <div className='note-form'>
            <form onSubmit={handleSubmit}>
                <input type='text' ref={ref} onChange={handleChange} value={value}/>
                <button>Add Note</button>
            </form>
        </div>
    )
}