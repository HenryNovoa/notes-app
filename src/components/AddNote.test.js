import { shallow, mount } from 'enzyme'
import React, { useContext, useReducer } from 'react'
import { AddNote } from './AddNote'
import { NotesContext } from '../context'
import { notesReducer } from '../reducer'
import toJson from 'enzyme-to-json'



describe('AddNote Component', ()=> {
    it('expect to render AddNote component', ()=> {
       const wrapper = mount(<NotesContext.Provider value={{state: {
        currentNote: null,
        notes: [
            {id: 1, text: 'Do small assestment'},
            {id: 2, text: 'Write Tests'}
        ]
    }}}>
           <AddNote/>
       </NotesContext.Provider>)

    const promise = () => {
    return new Promise(resolve => {
      setTimeout(() => {
        wrapper.update()
        resolve(wrapper)
      }, 3000)
    })
  }
   

    return promise().then((res) => {
        expect(toJson(res)).toMatchSnapshot();
      });
    })

    it('expects to handle change correctly', ()=> {
        const wrapper = mount(<NotesContext.Provider value={{state: {
         currentNote: null,
         notes: [
             {id: 1, text: 'Do small assestment'},
             {id: 2, text: 'Write Tests'}
         ]
     }}}>
            <AddNote/>
        </NotesContext.Provider>)
 
     const promise = () => {
     return new Promise(resolve => {
       setTimeout(() => {
         wrapper.update()
         resolve(wrapper)
       }, 3000)
     })
   }
    
 
     return promise().then((res) => {
         console.log({wrapper})
         console.log({res})
         wrapper.find('input').simulate('change', {target: {
             value: 'test'
         }})
         expect(wrapper.find('input').prop('value')).toEqual('test')
       });
     })
   
})