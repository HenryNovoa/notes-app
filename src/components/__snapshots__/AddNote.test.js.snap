// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote Component expect to render AddNote component 1`] = `
<AddNote>
  <div
    className="note-form"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button>
        Add Note
      </button>
    </form>
  </div>
</AddNote>
`;
